### DISCUSSION TITLE
USB storage mode

### DISCUSSION THREAD

**Discussion by dulanjank** [BimmerGestalt/AAIdrive#792] [Q&A]
Hi everyone, I keep getting a USB connected in storage mode issue in the app (I've switched been various USB options and currently in the Android Auto mode - I have a Samsung s24 but same issue in my note 20). Things seem to work fine via Bluetooth and album art works fine. What I'd like to have is USB audio for quality (have the b&w audio too). Would love to get it working. 
As an aside someone was saying they could get Google maps running. I have a 2019 640i (G32) - is this possible? 

Thank you!

**Comment by hufman** [comment]
USB app support in ID5+ and MyBMW is flaky, and my car recently stopped detecting the apps connection over USB entirely.
The car's USB Audio support was removed from Android Oreo in 2017, and I haven't heard of any workaround.
The Screen Mirroring addon lets you view your phone screen in the car, which can be combined with navigating with Google Maps on your phone. However, there is no touch screen support in the car app protocol.
The gmaps variant of AAIdrive merely uses Google Mapview for rendering and place searching and routing, it doesn't actually run Google Maps. It is against the Google Maps Terms of Service to show a Mapview in the car, so AAIdrive cannot provide builds of this feature.
